lib/logger.js

```diff
@@ -103,9 +105,37 @@ class Logger extends EventEmitter {
     let mac = undefined
     let ip = undefined
 
+    if (has(options, 'levels')) {
+      const type = typeOf(options.levels)
+      if (type !== 'array') {
+        const err = new TypeError('levels must be an array')
+        err.meta = {
+          got: type
+        }
+        throw err
+      }
+      const custom_levels = []
+      const level_regex = /^[A-Za-z]+$/
```

Comment:
This should be hoisted as a module level constant

Reply:
It's a constructor-only var, but I suppose if multiple instances are created, it'd be beneficial higher up.